lib/statsd.js

```diff
@@ -447,9 +483,14 @@ Client.prototype._close = function (callback) {
   }
 
   try {
-    if (this.protocol === 'tcp') {
+    if (this.protocol === TCP) {
       this.socket.destroy();
-    } else {
+    } else if (this.protocol === UDS) {
+      this.socket.close();
```

Comment:
Just for my understanding - since it's synchronous, this won't trigger the `close` event for the socket and trigger the callback twice right?

Reply:
It's because it's synchronous and not set up properly to emit the close event- the tests were hanging without this.  I'll clear up the comment below.